src/List/ItemList.js

```diff
@@ -82,6 +86,187 @@ const Items = props => {
     }
   };
 
+  //comparison method for sorting by purchase date.
+  const comparison = (itemA, itemB) => {
```

Comment:
A fun way to shorten this function is to just return `itemB.nextPurchaseDate - itemA.nextPurchaseDate`. It will either return a negative number, 0, or a positive number:
```js
function comparison(a, b) {
  return b.nextPurchaseDate - a.nextPurchaseDate;
}
```